src/logging.cpp

```diff
@@ -76,6 +76,14 @@ bool BCLog::Logger::StartLogging()
     return true;
 }
 
+void BCLog::Logger::DisconnectTestLogger()
+{
+    std::lock_guard<std::mutex> scoped_lock(m_cs);
+    m_buffering = true;
+    if (m_fileout != nullptr) fclose(m_fileout);
+    m_fileout = nullptr;
```

Comment:
This is causing the mac build failure, which is weird, maybe do 
LOCK(m_cs); 
instead of 
std::lock_guard<std::mutex> scoped_lock(m_cs);

Thoughts @UdjinM6?

Otherwise everything looks good to me